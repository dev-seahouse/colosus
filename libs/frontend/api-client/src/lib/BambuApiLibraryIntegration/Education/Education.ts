import BambuApiLibraryIntegrationBaseApi from '../_Base/Base';
import type {
  IBambuApiLibraryCalculateUniversityGoalAmountRequestDto,
  IBambuApiLibraryCalculateUniversityGoalAmountResponseDto,
} from '@bambu/shared';
import {
  EnumBambuApiLibraryCalculateUniversityGoalAmountRequestSpecialisation,
  EnumBambuApiLibraryCalculateUniversityGoalAmountRequestUniversityType,
} from '@bambu/shared';

export type CalculateUniversityGoalAmountRequestDto =
  IBambuApiLibraryCalculateUniversityGoalAmountRequestDto;

export type CalculateUniversityGoalAmountResponseDto =
  IBambuApiLibraryCalculateUniversityGoalAmountResponseDto;

export { EnumBambuApiLibraryCalculateUniversityGoalAmountRequestUniversityType as CalculateUniversityGoalAmountUniversityTypeEnum };

export { EnumBambuApiLibraryCalculateUniversityGoalAmountRequestSpecialisation as CalculateUniversityGoalAmountSpecialisation };

export class BambuApiLibraryIntegrationEducationApi extends BambuApiLibraryIntegrationBaseApi {
  constructor(private readonly apiPath = 'education-apis') {
    super();
  }

  /**
   * Provide access to the Calculate Education Goal Amount API in Bambu API Library
   * - {@link http://localhost:9000/openapi#/Bambu%20API%20Library/CalculateUniversityGoalAmount}.
   */
  public async calculateUniversityGoalAmount(
    req: CalculateUniversityGoalAmountRequestDto
  ) {
    return await this.axios.post<CalculateUniversityGoalAmountResponseDto>(
      `${this.apiPath}/calculate-university-goal-amount`,
      req
    );
  }
}

export default BambuApiLibraryIntegrationEducationApi;
