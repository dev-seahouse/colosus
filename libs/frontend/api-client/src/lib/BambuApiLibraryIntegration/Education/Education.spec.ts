import BambuApiLibraryIntegrationEducationApi, {
  CalculateUniversityGoalAmountSpecialisation,
} from './Education';
import { CalculateUniversityGoalAmountUniversityTypeEnum } from './Education';

describe('Education', () => {
  const bambuApiLibraryIntegrationEducationApi =
    new BambuApiLibraryIntegrationEducationApi();
  const mockRequest = {
    age: 16,
    ageOfUni: 18,
    maxGoalYear: 24,
    universityType: CalculateUniversityGoalAmountUniversityTypeEnum.PUBLIC,
    country: 'US',
    inflationRate: 0.03,
    currentYear: 2018,
    specialisation: CalculateUniversityGoalAmountSpecialisation.GENERAL,
    state: 'Delaware',
    residencyType: 'instate',
  };

  describe('calculateUniversityGoalAmount', () => {
    it('should return a valid response', async () => {
      const res =
        await bambuApiLibraryIntegrationEducationApi.calculateUniversityGoalAmount(
          mockRequest
        );
      expect(res.data).toMatchSnapshot();
    });
  });
});
