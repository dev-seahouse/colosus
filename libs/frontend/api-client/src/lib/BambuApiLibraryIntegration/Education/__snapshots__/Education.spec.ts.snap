// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Education > calculateUniversityGoalAmount > should return a valid response 1`] = `
{
  "currency": "USD",
  "educationCost": 155502,
  "goalYear": 2020,
  "maxYearsToGoal": 8,
  "universityCost": 146576,
  "yearsToGoal": 2,
}
`;
